### PR TITLE
Feature msg business consumer

### PR DESCRIPTION
**Pull request by kebingxing**
### support trace the real result of kafka message processing at the business side
- [x] If this is non-trivial feature, paste the links/URLs to the design doc.
- [x] Update the documentation to include this new feature.
- [x] Tests(including UT, IT, E2E) are added to verify the new feature.
- [x] If it's UI related, attach the screenshots below.

- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<no>.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking-java/blob/main/CHANGES.md).

### instruction
Previously, kafka and other message consumers could only trace the messages pulled by the clients, but the actual processing results of the business could not be traced. Our project had a large number of message processing scenarios, Tracking business message processing results can improved our troubleshooting efficiency~
1.before:
image
2. after
image
After the message is processed by the business side, can customize record the processing results
-- methods as follows
TraceMsgContext.ConsumerMsgSucceed
TraceMsgContext.ConsumerMsgFailure
image

It is my honor to contribute to the skywalking. Thank you for taking the time to read it, and I hope you can give me some suggestions. If there is any mistake, I will adjust it in time

### CODE REVIEW

File: apm-sniffer/apm-sdk-plugin/kafka-plugin/src/main/java/org/apache/skywalking/apm/plugin/kafka/KafkaConsumerInterceptor.java [hunk 1]
```diff
@@ -44,6 +48,17 @@
 
     public static final String OPERATE_NAME_PREFIX = "Kafka/";
     public static final String CONSUMER_OPERATE_NAME = "/Consumer/";
+    private static final Gson GSON = new Gson();
+
+    private static boolean TRACE_MSG_OPEN;
+
+    public KafkaConsumerInterceptor() {
+        try {
+            Class.forName("org.apache.skywalking.apm.toolkit.trace.msg.TraceMsg");
```
Comment (wu-sheng) [inline apm-sniffer/apm-sdk-plugin/kafka-plugin/src/main/java/org/apache/skywalking/apm/plugin/kafka/KafkaConsumerInterceptor.java:57]: This is totally wrong.

**Review by wu-sheng — CHANGES_REQUESTED** [review]
If you want to propose a new API and a new feature, you should submit a discussion first. 
Your changes here are not acceptable for upstream, and wrong from tech perspective too.

File: apm-sniffer/apm-sdk-plugin/kafka-plugin/src/main/java/org/apache/skywalking/apm/plugin/kafka/KafkaConsumerInterceptor.java (hunk 1 above)
Comment (kebingxing) [reply apm-sniffer/apm-sdk-plugin/kafka-plugin/src/main/java/org/apache/skywalking/apm/plugin/kafka/KafkaConsumerInterceptor.java:57]: Thank you for your guidance.
The original target here, The KafkaConsumerInterceptor can automatically stores trace information into the message that are implemented the TraceMsg interface.

I will optimize this code;  The pseudo-code used is as follows:
1.  String trace=TraceMsgContext.getCurretTrace();    //provided to obtain trace information
2.  msg.setTraceData(trace);  // customized trace data  in message body
3. TraceMsgContext.ConsumerMsgSucceed(msg.getTraceData()) //trace the real result of kafka message processing

In addition, for the scenarios supported by this feature, whether it is valuable~? 
I submit a discussion  later;

Thank you again for your guidance; and Thank SkyWalking for bringing a lot of value to the project;

Comment (wu-sheng) [reply apm-sniffer/apm-sdk-plugin/kafka-plugin/src/main/java/org/apache/skywalking/apm/plugin/kafka/KafkaConsumerInterceptor.java:57]: Please submit a discussion to sync the case you faced and try to resolve.
Then we could move on tech details, finally as a pull request.
Right now, I am confused with this PR.

**Pull request closed without merging**